package.json

```diff
@@ -97,5 +101,26 @@
       "@babel/plugin-transform-regenerator",
       "@babel/plugin-transform-runtime"
     ]
+  },
+  "commitlint": {
+    "extends": [
+      "@commitlint/config-conventional"
+    ],
+    "rules": {
+      "scope-case": [
+        2,
+        "always",
+        [
+          "lower-case",
+          "pascal-case"
+        ]
+      ]
+    }
+  },
+  "husky": {
+    "hooks": {
+      "pre-commit": "npm run test:eslint",
+      "commit-msg": "commitlint -E HUSKY_GIT_PARAMS"
```

Comment:
A bit out of scope, but is there a reason why we're not running tests `pre-push`?

Reply:
with github already running the tests for every push, i find it a bit pointless to have this script. with that being said, I don't mind adding a `pre-push` script. but not yet at least. maybe after adding some tests to this repo.

Reply:
I don’t think it’s pointless for the `pre-push` commit to run. Having the developer running tests and making sure they work before pushing to Github (specially if a PR is open) would save us from running unnecessary build in our CI. Freeing up resources is important as the project grows. It would be a good standard across all our projects

Reply:
will add it when we come with the testing environment.